2024.1/bulk/repo-access-api.adoc

```diff
@@ -0,0 +1,526 @@
+include::../../shared/issue-footnotes.adoc[]
+:serialization: ../serialization/serialization
+:m3: ../metametamodel/metametamodel
+:trash: trash-api
+:arch: ../reference-architecture/reference-architecture
+:chunk: <<{serialization}.adoc#SerializationChunk, SerializationChunk>>
+
+[[bulk]]
+= Bulk Repository Access API
+:toc: preamble
+:toclevels: 2
+
+The bulk API *version 2024.1* is used to store and retrieve nodes in batches at the moment of invocation.{fn-org25}
+It is intended for CRUD operations on (larger) sets of nodes.
+it is *not* intended as a delta-oriented API that takes "modification commands" as arguments.
+
+== Conventions used in this document
+* ALL-CAPS key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in https://datatracker.ietf.org/doc/html/bcp14[BCP14] (https://datatracker.ietf.org/doc/html/rfc2119[RFC2119], https://datatracker.ietf.org/doc/html/rfc8174[RFC8174]) when, and only when, they appear in all capitals, as shown here.
+* Footnotes refer to more discussions and rationale, but are non-normative.
+
+[[use-cases]]
+== Use Cases
+We describe the use cases from a client's perspective.
+
+* "I'm starting up, which partitions do you know about?", i.e. <<listPartitions, list possible contents>>.
+* "Let's have a new slate, add a new partition", i.e. <<createPartitions, partition creation>>.
+* "This part is not needed any more, remove the partition", i.e. <<deletePartitions, partition deletion>>.
+* "Give me the whole subtree under this node", i.e. actual <<retrieve, bulk data retrieval>>
+* "Here is a node forest, store it as I hand it to you", i.e. actual <<store, bulk data storage>>.
+** Sub-case with same behavior: "I'm a very simple client who knows nothing about deltas, so whatever I send you is the truth", i.e. <<store, click-save-button>>.
+* "I'm starting up, get me the current partition contents", i.e. <<retrieve, initial load>>.
+* "I want to create a guaranteed new node, which id to use?", i.e. retrieval of <<ids, unused ids>>.
+
+[[validity]]
+== Validity assumptions
+We don't need to know anything about languages for this API.{fn-org203}
```

Comment:
It would be a pretty long bullet list (the whole section), thus I'm reluctant.
We could add some explanation, but according to https://diataxis.fr/reference/ we should not do that in reference docs.